bundles/org.openhab.binding.networkupstools/src/main/java/org/openhab/binding/networkupstools/internal/NUTHandler.java

```diff
@@ -0,0 +1,404 @@
+     */
+    private @Nullable Map<String, String> retrieveVariables() {
+        logger.trace("Get NUT variables from server for thing: {}({})", thing.getLabel(), thing.getUID());
+        final NutApi localNutApi = nutApi;
+
+        if (localNutApi == null) {
+            return null;
+        }
+        return wrappedNutApiCall(localNutApi::getVariables);
+    }
+
+    /**
+     * Convenience method that wraps the call to the api and handles exceptions.
+     *
+     * @param <T> Return type of the call to the api
+     * @param nutApiFunction function that will be called
+     * @return the value returned by the api call or null in case of an error
+     */
+    private <T> T wrappedNutApiCall(final NutFunction<String, T> nutApiFunction) {
+        try {
+            final NUTConfiguration localConfig = config;
+
+            if (localConfig == null) {
+                return null;
+            }
+            return nutApiFunction.apply(localConfig.device);
+        } catch (final NutException e) {
+            logger.debug("Refresh Network UPS Tools failed: ", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Updates the thing properties when the cache {@link #refreshPropertiesCache} is expired.
+     *
+     * @return returns true if success and false on error
+     */
+    private Boolean updateProperties() {
+        try {
+            final Map<String, String> variables = variablesCache.getValue();
+
+            if (variables != null) {
+                final Map<String, String> properties = editProperties();
+
+                for (final Parameters param : NUTBindingConstants.Parameters.values()) {
+                    final String value = deviceToString(variables, param.getNutName());
+
+                    if (value != null) {
+                        properties.put(param.getNutName(), value);
+                    }
+                }
+                updateProperties(properties);
+            }
+            return Boolean.TRUE;
+        } catch (final RuntimeException e) {
+            logger.debug("Updating parameters failed: ", e);
+            return Boolean.FALSE;
+        }
+    }
+
+    private void updateChannel(final Channel channel, @Nullable final Map<String, String> variables) {
+        try {
+            if (variables == null) {
+                return;
+            }
+            final State state;
+            final String id = channel.getUID().getId();
+            final NutName fixedChannel = NutName.channelIdToNutName(id);
+
+            if (fixedChannel == null) {
+                state = getDynamicChannelState(channel, variables);
+            } else {
+                state = fixedChannel.toState(variables);
+            }
+            updateState(channel.getUID(), state);
+        } catch (final NutException | RuntimeException e) {
```

Comment:
Is it needed to catch the `RuntimeException` here? Can you catch something more specific? Should you handle it more extended instead of only debug logging?

Reply:
It's a safety catch here. If something unexpected happens with updating a channel due to problems with matching the variable it won't block updating the other channels.